### PR TITLE
Feat: Multisig details

### PR DESCRIPTION
**Pull request by tuul-wq**
- Multisig details
image
image

### CODE REVIEW

File: src/renderer/entities/wallet/ui/ContactItem/ContactItem.tsx
Comment (tuul-wq) [inline src/renderer/entities/wallet/ui/ContactItem/ContactItem.tsx:1]: This component is very similar to Address components
In future we definitely must come to less address components [(task)](https://app.clickup.com/t/8692z05p3)

File: src/renderer/entities/wallet/ui/MultiAccountsList/MultiAccountsList.tsx [hunk 1]
```diff
@@ -17,8 +17,8 @@ export const MultiAccountsList = ({ accounts, className }: Props) => {
   const { t } = useI18n();
 
   return (
-    <>
-      <div className="flex mx-3 py-4">
+    <div className="flex flex-col">
+      <div className="flex px-5 py-4">
```
Comment (tuul-wq) [inline src/renderer/entities/wallet/ui/MultiAccountsList/MultiAccountsList.tsx:21]: Checked all places where this is being used

File: src/renderer/features/wallets/SelectWalletPairing/ui/SelectWalletPairing.tsx
Comment (tuul-wq) [inline src/renderer/features/wallets/SelectWalletPairing/ui/SelectWalletPairing.tsx:1]: Same order as in `<WalletSelect />`

File: src/renderer/shared/core/types/utility.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
 export type PartialBy<T, K extends keyof T> = Omit<T, K> & Pick<Partial<T>, K>;
 
+export type KeysOfType<T, TCondition> = {
```
Comment (tuul-wq) [inline src/renderer/shared/core/types/utility.ts:3]: Handy utility type
It grabs all properties from an object that satisfies condition 
Example:
```javascript
  KeysOfType<AbstractAccount, string> ==> 'name' | 'type'
```

File: src/renderer/shared/lib/utils/arrays.ts [hunk 3]
```diff
@@ -8,3 +10,29 @@
...
+
+/**
+ * Create dictionary with given key and value
+ * Keys can only be type of string, number or symbol
+ * @param collection array of items
+ * @param property field to be used as key
+ * @param predicate transformer function
+ * @return {Object}
+ */
+export function dictionary<T extends Record<K, PropertyKey>, K extends KeysOfType<T, PropertyKey>>(
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/arrays.ts:22]: Same as `keyBy` from _lodash_ except it has TS support for valid keys (`string | number | symbol` only)
and provide optional predicate function if we require some transformed value

File: src/renderer/shared/ui/Modals/BaseModal/BaseModal.tsx [hunk 4]
```diff
@@ -61,7 +58,6 @@ const BaseModal = ({
                   name="close"
                   size={20}
                   className="absolute top-[18px] right-[14px] z-10"
-                  ariaLabel={t('basemodal.closeButton')}
```
Comment (tuul-wq) [inline src/renderer/shared/ui/Modals/BaseModal/BaseModal.tsx:64]: I guess we can live without it for a while 😄 
`const { t } = useI18n();` was wrong inside pure UI component

**Review by tuul-wq — COMMENTED** [review]
My comments

**Comment by tuul-wq** [comment]
Task linked: [CU-862kcvq47 Show Multisig wallet details](https://app.clickup.com/t/862kcvq47)

File: src/renderer/shared/core/types/utility.ts (hunk 2 above)
Comment (egor0798) [reply src/renderer/shared/core/types/utility.ts:3]: Nice!

File: src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts [hunk 5]
```diff
@@ -17,18 +18,83 @@ const $accounts = combine(
   },
 );
 
-const $connected = combine($accounts, walletConnectModel.$client, (accounts, client): boolean => {
-  const account = accounts[0];
-  if (!client || !account || !accountUtils.isWalletConnectAccount(account)) return false;
+const $singleShardAccount = combine(
+  {
+    accounts: walletModel.$accounts,
+  },
```
Comment (Asmadek) [inline src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts:24]: Is object required here? You can just provide `walletModel.$accounts` and use it as single argument in callback

**Review by Asmadek — APPROVED** [review]

**Review by egor0798 — APPROVED** [review]

**Pull request merged**